### ISSUE TITLE
The issue encountered when running “d_adapt.py”

### ISSUE THREAD

**Issue by LTT-5** [thuml/Transfer-Learning-Library#272]
Hello, thank you very much for your excellent work! 

I downloaded the code and tried running it, but I encountered some issues. 
I used both _cycle_gan.py_ and _source_only.py_ to obtain the _model_final.pth_ file, and then I ran the following code.

**Code:**
python d_adapt.py --workers-c 1 --max-train-c 1 --ignored-scores-c 0.05 0.5 --config-file d_adapt/config/faster_rcnn_R_101_C4_cityscapes.yaml -s Cityscapes datasets_bymyself/cityscapes_in_voc -t FoggyCityscapes datasets_bymyself/foggy_cityscapes_in_voc/ --test FoggyCityscapesTest datasets_bymyself/foggy_cityscapes_in_voc/ --finetune --trade-off 0.5 --bbox-refine OUTPUT_DIR logs/faster_rcnn_R_101_C4/cityscapes2foggy/phase1 MODEL.WEIGHTS logs/cyclegan/faster_rcnn_R_101_C4/cityscapes2foggy/model_final.pth SEED 0

I changed --max-train-c to 1, but I noticed that the code ran to the 8th epoch and kept going. Also, I am using my modified dataset (i.e., folder "datasets_bymyself"), which contains only 33 test images, but the results show 159 images. Could you please help me solve these issues? I look forward to your feedback.

**Running:**
Epoch: [7][   0/1000]   Time 0.4 (0.4)  Data 0.0 (0.0)  Loss 0.34 (0.34)        Loss(t) 0.00 (0.00)     Trans Loss 0.68 (0.68)  Cls Acc 100.0 (100.0)  Domain Acc 53.9 (53.9)
Epoch: [7][ 100/1000]   Time 3.2 (3.2)  Data 2.7 (2.7)  Loss 0.34 (0.34)        Loss(t) 0.00 (0.00)     Trans Loss 0.68 (0.68)  Cls Acc 100.0 (99.9)   Domain Acc 53.1 (55.7)
Epoch: [7][ 200/1000]   Time 11.5 (3.3) Data 10.8 (2.8) Loss 0.34 (0.34)        Loss(t) 0.00 (0.00)     Trans Loss 0.68 (0.68)  Cls Acc 100.0 (99.9)   Domain Acc 50.8 (55.8)
Epoch: [7][ 300/1000]   Time 3.2 (3.3)  Data 2.7 (2.9)  Loss 0.34 (0.34)        Loss(t) 0.00 (0.00)     Trans Loss 0.67 (0.68)  Cls Acc 100.0 (99.9)   Domain Acc 60.2 (55.3)
Epoch: [7][ 400/1000]   Time 3.2 (3.3)  Data 2.8 (2.8)  Loss 0.33 (0.34)        Loss(t) 0.00 (0.00)     Trans Loss 0.65 (0.68)  Cls Acc 100.0 (99.9)   Domain Acc 60.9 (55.3)
Epoch: [7][ 500/1000]   Time 3.0 (3.3)  Data 2.6 (2.9)  Loss 0.33 (0.34)        Loss(t) 0.00 (0.00)     Trans Loss 0.66 (0.68)  Cls Acc 100.0 (99.9)   Domain Acc 59.4 (55.2)
Epoch: [7][ 600/1000]   Time 3.2 (3.3)  Data 2.7 (2.8)  Loss 0.34 (0.34)        Loss(t) 0.00 (0.00)     Trans Loss 0.68 (0.68)  Cls Acc 100.0 (99.9)   Domain Acc 55.5 (55.4)
Epoch: [7][ 700/1000]   Time 3.2 (3.3)  Data 2.8 (2.8)  Loss 0.34 (0.34)        Loss(t) 0.00 (0.00)     Trans Loss 0.68 (0.68)  Cls Acc 100.0 (99.9)   Domain Acc 56.2 (55.4)
Epoch: [7][ 800/1000]   Time 3.1 (3.3)  Data 2.7 (2.9)  Loss 0.35 (0.34)        Loss(t) 0.00 (0.00)     Trans Loss 0.69 (0.68)  Cls Acc 100.0 (99.9)   Domain Acc 50.0 (55.4)
Epoch: [7][ 900/1000]   Time 3.2 (3.3)  Data 2.8 (2.8)  Loss 0.34 (0.34)        Loss(t) 0.00 (0.00)     Trans Loss 0.69 (0.68)  Cls Acc 100.0 (99.9)   Domain Acc 51.6 (55.5)
Test: [  0/159] Time  8.105 ( 8.105)    Loss 1.0151e+00 (1.0151e+00)    Acc@1  87.50 ( 87.50)
Test: [100/159] Time  3.051 ( 3.023)    Loss 7.1992e-04 (5.2414e-01)    Acc@1 100.00 ( 91.29)
 * Acc@1 94.444
global correct: 94.4
mean correct:71.7
mean IoU: 63.1

**Environment:**
Windows+Python3.8+torch1.10.1
